src/httprpc.cpp

```diff
@@ -250,14 +250,24 @@ static bool InitRPCAuthentication()
         LogPrintf("Config options rpcuser and rpcpassword will soon be deprecated. Locally-run instances may remove rpcuser to use cookie-based auth, or may be replaced with rpcauth. Please see share/rpcauth for rpcauth auth generation.\n");
         strRPCUserColonPass = gArgs.GetArg("-rpcuser", "") + ":" + gArgs.GetArg("-rpcpassword", "");
     }
-    if (gArgs.GetArg("-rpcauth","") != "")
+    if (gArgs.GetArg("-rpcauth", "") != "" || gArgs.GetArg("-rpcauthfile", "") != "")
     {
         LogPrintf("Using rpcauth authentication.\n");
         for (std::string rpcauth : gArgs.GetArgs("-rpcauth")) {
             std::vector<std::string> fields;
             boost::split(fields, rpcauth, boost::is_any_of(":$"));
             if (fields.size() == 3) g_rpcauth.push_back(fields);
         }
+        for (std::string path : gArgs.GetArgs("-rpcauthfile")) {
+            fsbridge::ifstream file;
+            file.open(path);
+            if (!file.is_open()) continue;
+            std::string rpcauth;
+            std::getline(file, rpcauth);
+            std::vector<std::string> fields;
+            boost::split(fields, rpcauth, boost::is_any_of(":$"));
+            if (fields.size() == 3) g_rpcauth.push_back(fields);
```

Comment:
> I think this should refuse to load

You mean the process should exit?

Reply:
re: https://github.com/bitcoin/bitcoin/pull/20407#discussion_r526080624

> > I think this should refuse to load
> 
> You mean the process should exit?

I would exit, but don't think it would be crazy not to exit, do think it would be crazy to silently ignore a configuration error giving no feedback. I am presuming returning false exits, but didn't check.

Reply:
See above, it's not necessarily an error

Reply:
@luke-jr please review #20461, this one should rebase with that.